src/Mollie/WC/Helper/ApplePayDirectHandler.php

```diff
@@ -0,0 +1,1180 @@
+                    'needShipping' => $cart->needs_shipping(),
+                    'subtotal' => $cart->get_subtotal(),
+
+                ],
+                'shop' => [
+                    'countryCode' => $shopCountryCode,
+                    'currencyCode' => $currencyCode,
+                    'totalLabel' => $totalLabel
+                ],
+                'ajaxUrl' => admin_url('admin-ajax.php')
+            ];
+        }
+        return [];
+    }
+
+    /**
+     * Method to validate the merchant against Apple system through Mollie
+     * On fail triggers and option that shows an admin notice showing the error
+     * On success returns the validation data to the script
+     */
+    public function validateMerchant()
+    {
+        $applePayRequestDataObject = $this->applePayDataObjectHttp();
+        $applePayRequestDataObject->validationData($_POST);
+
+        if (!$this->isNonceValid($applePayRequestDataObject)) {
+            wp_send_json_error('no nonce');
+        }
+        $validationUrl = $applePayRequestDataObject->validationUrl;
+        $completeDomain = get_site_url();
+        $removeHttp = ["https://", "http://", "/"];
+        $domain = str_replace($removeHttp, "", $completeDomain);
+
+        try {
+            $json = $this->validationApiWalletsEndpointCall(
+                $domain,
+                $validationUrl
+            );
+        } catch (\Mollie\Api\Exceptions\ApiException $exc) {
+            update_option('g_test', $exc->getMessage());
+            update_option('mollie_wc_applepay_validated', 'no');
+            wp_send_json_error($exc->getMessage());
+        }
+
+        update_option('mollie_wc_applepay_validated', 'yes');
+        update_option('mollie_apple_pay_session_response', $json);
+
+        wp_send_json_success($json);
+    }
+
+    /**
+     * Method to validate and update the shipping contact of the user
+     * It updates the amount paying information if needed
+     * On error returns an array of errors to be handled by the script
+     * On success returns the new contact data
+     */
+    public function updateShippingContact()
+    {
+        $applePayRequestDataObject = $this->applePayDataObjectHttp();
+        $applePayRequestDataObject->updateContactData($_POST);
+
+        if (!$this->isNonceValid($applePayRequestDataObject)) {
+            return;
+        }
+        if ($applePayRequestDataObject->hasErrors()) {
+            $this->responseWithDataErrors($applePayRequestDataObject->errors);
+            return;
+        }
+
+        if (class_exists('WC_Countries')) {
+            $countries = $this->createWCCountries();
+            $allowedSellingCountries = $countries->get_allowed_countries();
+            $allowedShippingCountries = $countries->get_shipping_countries();
+            $isAllowedSellingCountry = array_key_exists(
+                $applePayRequestDataObject->simplifiedContact['country'],
+                $allowedSellingCountries
+            );
```

Comment:
Always check if key exists before accessing to a value. Talking about `$applePayRequestDataObject->simplifiedContact['country']`. And I would move it in a variable since you're consuming it multiple times.

Reply:
Yeah, this value is coming from the $applePayRequestDataObject that lists the errors if something is missing and so before we reach that point there is a check for the $applePayRequestDataObject->hasErrors() and returns with the errors to Apple so the user can correct it. So I guess if I checked there, there's no need to check again?